examples_tests/06.MeshLoaders/main.cpp

```diff
@@ -51,22 +51,50 @@ int main()
     auto mesh = meshes_bundle.getContents().first[0];
     auto mesh_raw = static_cast<asset::ICPUMesh*>(mesh.get());
 
+    //we can safely assume that all meshbuffers within mesh loaded from OBJ has same DS1 layout (used for camera-specific data)
+    //so we can create just one DS
     asset::ICPUDescriptorSetLayout* ds1layout = mesh_raw->getMeshBuffer(0u)->getPipeline()->getLayout()->getDescriptorSetLayout(1u);
-    //pipelines attached to meshbuffers from OBJ loader has DS1 layout being "irr/builtin/ds_layout/default_ds1_layout"
-    //that is designed for use with "irr/builtin/ds/default_ds1", so let's use it!
-    auto ds1_bundle = am->getAsset("irr/builtin/descriptor_set/basic_view_parameters", lp);
-    //you might want to IAsset::clone() this DS in order to have exact copy but not the same object. However in this case won't be doing it.
-    //(same CPU object -> same GPU object returned from driver->getGPUObjectsFromAssets())
-    //use IAsset::clone()'s optional parameter to clone the buffer (being the only descriptor in default DS1) as well
-    auto ds1 = ds1_bundle.getContents().first[0];
+    uint32_t ds1UboBinding = 0u;
+    for (const auto& bnd : ds1layout->getBindings())
+        if (bnd.type==asset::EDT_UNIFORM_BUFFER)
+        {
+            ds1UboBinding = bnd.binding;
+            break;
+        }
+
+    constexpr size_t STD140_ROW_MAJOR_MVP_SZ = sizeof(float)*16ull;
+    constexpr size_t STD140_ROW_MAJOR_MV_SZ = sizeof(float)*12ull;
+    constexpr size_t STD140_ROW_MAJOR_NORMAL_MAT_SZ = sizeof(float)*12ull;
+
+    size_t neededDS1UBOsz = 0ull;
+    {
+        size_t matrixSz[asset::IPipelineMetadata::ECSI_COUNT]{};
+        matrixSz[asset::IPipelineMetadata::ECSI_WORLD_VIEW_PROJ] = STD140_ROW_MAJOR_MVP_SZ;
+        matrixSz[asset::IPipelineMetadata::ECSI_WORLD_VIEW] = STD140_ROW_MAJOR_MV_SZ;
+        matrixSz[asset::IPipelineMetadata::ECSI_WORLD_VIEW_INVERSE_TRANSPOSE] = STD140_ROW_MAJOR_NORMAL_MAT_SZ;
+        auto pipelineMetadata = static_cast<const asset::IPipelineMetadata*>(mesh_raw->getMeshBuffer(0u)->getPipeline()->getMetadata());
+        for (const auto& shdrIn : pipelineMetadata->getCommonRequiredInputs())
+            if (shdrIn.descriptorSection.type==asset::IPipelineMetadata::ShaderInput::ET_UNIFORM_BUFFER && shdrIn.descriptorSection.uniformBufferObject.set==1u && shdrIn.descriptorSection.uniformBufferObject.binding==ds1UboBinding)
+                neededDS1UBOsz = std::max(neededDS1UBOsz, shdrIn.descriptorSection.uniformBufferObject.relByteoffset+matrixSz[shdrIn.type]);
```

Comment:
maybe we should add `byteSize` to augment the `relByteOffset`  of the `shdrIn.descriptorSection.uniformBufferObject` ?

Then you wouldn't need all the matrix size constants above.